sdk/storage/Azure.Storage.Blobs.Batch/src/Models/StorageError.cs

```diff
@@ -41,9 +42,10 @@ internal partial class StorageError
         /// <summary>
         /// Create an exception corresponding to the StorageError.
         /// </summary>
+        /// <param name="clientDiagnostics">The <see cref="ClientDiagnostics"/> instance to use.</param>
         /// <param name="response">The failed response.</param>
         /// <returns>A RequestFailedException.</returns>
-        public Exception CreateException(Azure.Response response)
-            => StorageExceptionExtensions.CreateException(response, Message, null, Code, AdditionalInformation);
+        public Exception CreateException(ClientDiagnostics clientDiagnostics, Azure.Response response)
+            => clientDiagnostics.CreateRequestFailedExceptionWithContent(response, message: Message, content: null, errorCode: response.GetErrorCode(Code), additionalInfo: AdditionalInformation);
```

Comment:
Is there a specific reason you're switching from the response content's `<Code>` element to the `x-ms-error-code` response header?  I had to actually check to make sure this wouldn't confuse a batch sub-response, but this is only used from the generated code so it should be a safe change.

Reply:
I think I'm missing something but `response.GetErrorCode(Code)` would prefer `Code` over the header.